### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by aliok**
Verification: Follow the instructions in the updated README or checkout the CircleCI jobs (now we have 2 in the default workflow)

Note that the whole thing is to be squashed.

See the notes in JIRA: https://issues.jboss.org/browse/AEROGEAR-3606 about why some stuff is not tested

### CODE REVIEW

**Comment by aliok** [comment]
@darahayes please have a look at this PR after the demo

File: README.md [hunk 1]
```diff
@@ -56,10 +56,40 @@ Going to remove aerogeardatasyncserver_postgres_1
...
+npm run db:init
+npm run dev
+```
+
+In another session, run the tests:
+```
+npm run test:integration
+```
+
+### Running all tests with CircleCi CLI
```
Comment (darahayes) [inline README.md]: ❤️

File: integration_test/helper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  this.deleteConfig = async () => {
+    await qi.bulkDelete('DataSources')
+    await qi.bulkDelete('GraphQLSchemas')
+    await qi.bulkDelete('Resolvers')
+  }
+
+  this.feedConfig = async (configFile) => {
+    const config = require(`./config/${configFile}`)
+    if (!config.description) {
+      throw new Error(`Please define the description in file ./config/${configFile}`)
```
Comment (darahayes) [inline integration_test/helper.js]: I love this!

File: .circleci/config.yml [hunk 3]
```diff
@@ -35,20 +60,24 @@ workflows:
...
+            tags:
+              only: /.*/
+      - integration_test:
           filters:
             tags:
               only: /.*/
       - docker_push_master:
           requires:
-            - test
+            - unit_test
```
Comment (darahayes) [inline .circleci/config.yml]: can we also add `- integration_test` here

File: .circleci/config.yml [hunk 4]
```diff
@@ -35,20 +60,24 @@ workflows:
...
           filters:
             branches:
               only:
                 - master
       - docker_push_release:
           requires:
-            - test
+            - unit_test
```
Comment (darahayes) [inline .circleci/config.yml]: again we should add `- integration_test` here

**Comment by darahayes** [comment]
@aliok I've left one minor comment about running integration tests as part of the release steps. Other than that, this is fantastic!

**Comment by aliok** [comment]
Made integration tests part of "docker_push_*" jobs in the CircleCI config and merging this one now.

**Pull request merged**